internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go

```diff
@@ -343,3 +355,95 @@ func (k *KafkaManager) updateClusterStatusCapacityMaxMetric(c services.KafkaStre
+			orgBillingModelID := fmt.Sprintf("%s-%s", kafka.OrganisationId, kafka.ActualKafkaBillingModel)
+			active, ok := subscriptionStatusByOrgAndBillingModel[orgBillingModelID]
+			if !ok {
+				isActive, err := k.kafkaService.IsSubscriptionActive(kafka)
+				if err != nil {
+					svcErrors = append(svcErrors, errors.Wrapf(err, "failed to get subscription status of kafka instance %q", kafka.ID))
+					continue
+				}
+				subscriptionStatusByOrgAndBillingModel[orgBillingModelID] = *isActive
+				active = *isActive
+			}
+
+			if active {
+				// if subscription is active, set expires_at to null
+				if err := expiresAt.Scan(nil); err != nil {
+					svcErrors = append(svcErrors, errors.Wrapf(err, "failed to set expires_at value for kafka instance %q to NULL", kafka.ID))
+					continue
+				}
+			} else {
+				// if subscription is not valid and expires_at is not already set, set its value based on the current time and grace period allowance
+				if !kafka.ExpiresAt.Valid {
+					// set expires_at to now + grace period days
+					billingModel, err := k.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
+					if err != nil {
+						svcErrors = append(svcErrors, errors.Wrapf(err, "failed to get billing model of kafka instance %q", kafka.ID))
+						continue
+					}
+
+					duration, err := time.ParseDuration(fmt.Sprintf("%dh", billingModel.GracePeriodDays*24))
+					if err != nil {
+						svcErrors = append(svcErrors, errors.Wrapf(err, "failed to parse expiration duration for kafka instance %q", kafka.ID))
+						continue
+					}
+
+					if err := expiresAt.Scan(time.Now().Add(duration)); err != nil {
+						svcErrors = append(svcErrors, errors.Wrapf(err, "failed to set expires_at value for kafka instance %q", kafka.ID))
+						continue
+					}
+				}
+			}
+		} else {
+			// any Kafka instance types with lifespan seconds defined will have their expires_at value set on creation.
+			// this is only temporary to ensure that the expires_at value is set for any Kafka instances created during the rollout of this change.
+			if !kafka.ExpiresAt.Valid {
+				// set expires_at to created_at + lifespanSeconds
+				duration, err := time.ParseDuration(fmt.Sprintf("%ds", *instanceSize.LifespanSeconds))
+				if err != nil {
+					svcErrors = append(svcErrors, errors.Wrapf(err, "failed to parse expiration duration for kafka instance %q", kafka.ID))
+					continue
+				}
+
+				if err := expiresAt.Scan(kafka.CreatedAt.Add(time.Duration(duration))); err != nil {
+					svcErrors = append(svcErrors, errors.Wrapf(err, "failed to set expires_at value for kafka instance %q", kafka.ID))
+					continue
+				}
+			}
+		}
+
+		if expiresAt != kafka.ExpiresAt {
+			if err := k.kafkaService.Updates(kafka, map[string]interface{}{
+				"expires_at": expiresAt,
+			}); err != nil {
+				svcErrors = append(svcErrors, errors.Wrapf(err, "failed to update expires_at value for kafka instance %q", kafka.ID))
+			}
+		}
```

Comment:
I am not sure I understand it.

This will perform the update all the time instead of conditionally.

Reply:
Nope. When the expireAt is set, the current code exits and then the `if` checks if `expireAt` has been changed.
With this change we update it and continue to the next cluster instead of using a temporary `expireAt` and then checking if it has been changed.

Reply:
I think I understand the proposal now.

The method will perform the update all the time instead of conditionally, but the method will only be called under certain conditions in the blocks above.

Reply:
yup. exactly.